pkg/planner/core/optimizer.go

```diff
@@ -112,6 +112,7 @@ var optRuleList = []base.LogicalOptRule{
 	&PushDownSequenceSolver{},
 	&EliminateUnionAllDualItem{},
 	&ResolveExpand{},
+	&EmptySelectionEliminator{},
```

Comment:
better always keep ResolveExpand as the last rule

Reply:
Updated